system/jlib/jstats.cpp

```diff
@@ -1869,6 +1869,21 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
             cur.mergeInto(target);
     }
 
+    virtual void visit(IStatisticVisitor & visitor) const
+    {
+        if (visitor.visitScope(*this))
+        {
+            for (auto const & cur : children)
+                cur.visit(visitor);
+        }
+    }
+
+    virtual void visitChildren(IStatisticVisitor & visitor) const
```

Comment:
Is this function needed?

Reply:
It is needed if you ever wanted to implement a function that summarised statistics by level.  E.g., workflowid, graph, subgraph.  The visitor could track the change in scope and recurse itself.
This could be the mechanism for all visitors - but it will end up with an extra virtual call, and a deeper call stack.